src/main/java/org/opentripplanner/updater/spi/TimetableSnapshotFlush.java

```diff
@@ -0,0 +1,43 @@
+package org.opentripplanner.updater.spi;
+
+import org.opentripplanner.ext.siri.SiriTimetableSnapshotSource;
+import org.opentripplanner.updater.trip.TimetableSnapshotSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Flush the timetable snapshot buffer by committing pending changes.
+ * Exceptions occurring during the flush are caught and ignored: the scheduler can then retry
+ * the task later.
+ */
+public class TimetableSnapshotFlush implements Runnable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TimetableSnapshotFlush.class);
+
+  private final SiriTimetableSnapshotSource siriTimetableSnapshotSource;
+  private final TimetableSnapshotSource gtfsTimetableSnapshotSource;
+
+  public TimetableSnapshotFlush(
+    SiriTimetableSnapshotSource siriTimetableSnapshotSource,
+    TimetableSnapshotSource gtfsTimetableSnapshotSource
+  ) {
+    this.siriTimetableSnapshotSource = siriTimetableSnapshotSource;
+    this.gtfsTimetableSnapshotSource = gtfsTimetableSnapshotSource;
+  }
+
+  @Override
+  public void run() {
+    try {
+      LOG.debug("Flushing timetable snapshot buffer");
+      if (siriTimetableSnapshotSource != null) {
+        siriTimetableSnapshotSource.flushBuffer();
+      }
+      if (gtfsTimetableSnapshotSource != null) {
+        gtfsTimetableSnapshotSource.flushBuffer();
+      }
+      LOG.debug("Flushed timetable snapshot buffer");
+    } catch (Throwable t) {
```

Comment:
The `Throwable` opens up for a potensial critical error state, where the OTP Server is in a corrupt state, but does not go down. But the  pragmatic programmer in me tells me that we probably have bigger issues in the code, and that we should fokus on those. To fix it we would need to catch both RuntimeException and Throwable. Log both, but take down the application for Throwable. I think it is out-of-scope for this PR.

Reply:
I agree.

Is there ever a case where you can recover from an `OutOfMemoryError`? I think in such a case you really want to let the JVM die.

Reply:
This is a real case in a web server: a request thread uses up all memory and its thread dies, freeing up the memory again. We had this case in our production environment with some problematic GraphQL queries returning gigabytes of data. If the periodic task was running at the same time as such a request, the OOME could be triggered from the periodic task thread and make it stop for ever.

Reply:
See also Andrew's comment on that topic: https://github.com/opentripplanner/OpenTripPlanner/pull/5974#discussion_r1688066767

Reply:
There are two separate concerns here: recovering from errors, and logging errors. Indeed you probably want to just let certain severe errors run their course even if that incapacitates the server. But you nonetheless want to make sure they are logged so you know what happened. If I recall correctly, an error in a Runnable on an Executor can cause the executor thread to die, but silently. Then the tasks just pile up in a queue for no apparent reason, which would be very confusing to debug.

An OOME within the executor might kill the executor thread. You definitely want to log that, as it will presumably be followed by tasks piling up and the JVM crashing and you'd want to know why.